src/w3d/renderer/render2dsentence.cpp

```diff
@@ -207,8 +306,149 @@ void FontCharsClass::Update_Current_Buffer(int char_width)
     }
 }
 
+bool FontCharsClass::Locate_Font(const char *font_name, StringClass &font_file_path)
```

Comment:
Is this function often or rarely called? It looks expensive.

If this function is expensive and called many times, I suggest to build font path catalogue once, and then lookup from that catalogue afterwards. It should bring down computation time.